src/coreComponents/constitutive/contact/BartonBandis.hpp

```diff
@@ -127,12 +137,15 @@ GEOS_HOST_DEVICE
 GEOS_FORCE_INLINE
 real64 BartonBandisUpdates::computeHydraulicAperture( real64 const aperture,
                                                       real64 const normalTraction,
+                                                      integer const fractureState,
                                                       real64 & dHydraulicAperture_aperture,
                                                       real64 & dHydraulicAperture_dNormalStress ) const
 {
-  real64 const hydraulicAperture = ( aperture >= 0.0 ) ? (aperture + m_aperture0) : m_aperture0 / ( 1 + 9*normalTraction/m_referenceNormalStress );
-  dHydraulicAperture_dNormalStress = ( aperture >= 0.0 ) ? 0.0 : -hydraulicAperture / ( 1 + 9*normalTraction/m_referenceNormalStress ) * 9/m_referenceNormalStress;
-  dHydraulicAperture_aperture = ( aperture >= 0.0 ) ? 1.0 : 0.0;
+  using namespace fields::contact;
+
+  real64 const hydraulicAperture = ( fractureState == FractureState::Open ) ? (aperture + m_aperture0) : m_aperture0 / ( 1 + 9*LvArray::math::abs(normalTraction)/m_referenceNormalStress );
+  dHydraulicAperture_dNormalStress = ( fractureState == FractureState::Open ) ? 0.0 : -hydraulicAperture / ( 1 + 9*LvArray::math::abs(normalTraction)/m_referenceNormalStress ) * 9/m_referenceNormalStress;
+  dHydraulicAperture_aperture = ( fractureState == FractureState::Open ) ? 1.0 : 0.0;
```

Comment:
For current `HydroFractureSolver`, if the fracture is not open, this cannot be zero if we update `HydraulicAperture` implicitly using the current `aperture`. It should be something like `dHydraulicAperture_aperture = dHydraulicAperture_dNormalStress * dNormalStress_dAperture` where `dNormalStress_dAperture = penaltyStiffness`.

Reply:
For closed fractures, mechanical `aperture` is not included in the BartonBandis formula and hence zero.
Probably a following PR is needed to get it compatible with HydroFrac solver.